polaris/dataset/_dataset.py

```diff
@@ -348,14 +348,15 @@ def to_json(self, destination: str) -> str:
 
         # Copy over Zarr data to the destination
         if self.zarr_root is not None:
-            dest = zarr.open(zarr_archive, "w")
-            zarr.copy_all(source=self.zarr_root, dest=dest)
+            dest = zarr.DirectoryStore(zarr_archive)
+
+            zarr.copy_store(source=self.zarr_root.store.store, dest=dest)
```

Comment:
Could you try see if this works?

 ```suggestion
             zarr.copy_store(source=self.zarr_root.store, dest=dest)
```

I think it would, but maybe not!

Reply:
Ya I previously tried this and run into the following error:

```Traceback (most recent call last):
  File "/Users/larissa.troper/polaris/profiling.py", line 350, in <module>
    app()
  File "/Users/larissa.troper/anaconda3/envs/polaris/lib/python3.12/site-packages/typer/main.py", line 328, in __call__
    raise e
  File "/Users/larissa.troper/anaconda3/envs/polaris/lib/python3.12/site-packages/typer/main.py", line 311, in __call__
    return get_command(self)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/larissa.troper/anaconda3/envs/polaris/lib/python3.12/site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/larissa.troper/anaconda3/envs/polaris/lib/python3.12/site-packages/typer/core.py", line 716, in main
    return _main(
           ^^^^^^
  File "/Users/larissa.troper/anaconda3/envs/polaris/lib/python3.12/site-packages/typer/core.py", line 216, in _main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "/Users/larissa.troper/anaconda3/envs/polaris/lib/python3.12/site-packages/click/core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/larissa.troper/anaconda3/envs/polaris/lib/python3.12/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/larissa.troper/anaconda3/envs/polaris/lib/python3.12/site-packages/typer/main.py", line 683, in wrapper
    return callback(**use_params)  # type: ignore
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/larissa.troper/polaris/profiling.py", line 296, in profile
    profile_all(tmpdir, owner, name, n_bytes, n_cols, save_dir, fs, report=report)
  File "/Users/larissa.troper/polaris/profiling.py", line 189, in profile_all
    dataset.cache(os.path.join(tmpdir, "cache"))
  File "/Users/larissa.troper/polaris/polaris/dataset/_dataset.py", line 383, in cache
    self.to_json(self.cache_dir)
  File "/Users/larissa.troper/polaris/polaris/dataset/_dataset.py", line 353, in to_json
    zarr.copy_store(source=self.zarr_root.store, dest=dest)
  File "/Users/larissa.troper/anaconda3/envs/polaris/lib/python3.12/site-packages/zarr/convenience.py", line 732, in copy_store
    n_bytes_copied += buffer_size(data)
                      ^^^^^^^^^^^^^^^^^
  File "/Users/larissa.troper/anaconda3/envs/polaris/lib/python3.12/site-packages/zarr/util.py", line 380, in buffer_size
    return ensure_ndarray_like(v).nbytes
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/larissa.troper/anaconda3/envs/polaris/lib/python3.12/site-packages/numcodecs/compat.py", line 42, in ensure_ndarray_like
    mem = memoryview(buf)
          ^^^^^^^^^^^^^^^
TypeError: memoryview: a bytes-like object is required, not 'dict'```